libs/model/src/models/address.ts

```diff
@@ -144,6 +145,42 @@ export default (
       scopes: {
         withPrivateData: {},
       },
+      hooks: {
+        afterCreate: async (
+          address: AddressInstance,
+          options: Sequelize.CreateOptions<AddressAttributes>,
+        ) => {
+          // when address created, increment Community.address_count
+          const { community_id } = address;
+          if (!community_id) {
+            return;
+          }
+          const community = await models.Community.findByPk(community_id);
```

Comment:
Two things here (same argument for other increment/decrement hooks),

1. Shouldn't this be in the transaction? Otherwise the community can be deleted between this check and the increment
2. Maybe an easier way to do this would be to assume that the community exists and increment the address_count. Postgresql will return how many rows it updates. If it's 0, then the community didn't exist and we return. Otherwise the count is updated. This would clean up quite a few lines because we also wouldn't need to check that community_id exists.

Reply:
Good point. I've cleaned up the hooks to use a single query. Also added transactions around the original actions because it doesn't do so automatically.